src/AWS.Deploy.Recipes/CdkTemplates/AspNetAppAppRunner/Configurations/Configuration.cs

```diff
@@ -0,0 +1,120 @@
+
+namespace AspNetAppAppRunner.Configurations
+{
+    public class Configuration
+    {
+        /// <summary>
+        /// The Identity and Access Management Role that provides AWS credentials to the application to access AWS services
+        /// </summary>
+        public IAMRoleConfiguration ApplicationIAMRole { get; set; }
+
+        /// <summary>
+        /// The Identity and Access Management (IAM) role that provides gives the AWS App Runner service access to pull the container image from ECR.
+        /// </summary>
+        public IAMRoleConfiguration ServiceAccessIAMRole { get; set; }
+
+        /// <summary>
+        /// The name of the AppRunner service.
+        /// </summary>
+        public string ServiceName { get; set; }
+
+        /// <summary>
+        /// The port that your application listens to in the container. Defaults to port 80.
+        /// </summary>
+        public int Port { get; set; } = 80;
+
+        /// <summary>
+        /// An optional command that App Runner runs to start the application in the source image. If specified, this command overrides the Docker image’s default start command.
+        /// </summary>
+        public string? StartCommand { get; set; }
+
+        /// <summary>
+        /// The ARN of the KMS key that's used for encryption.
+        /// </summary>
+        public string? EncryptionKmsKey { get; set; }
+
+        /// <summary>
+        /// The number of consecutive checks that must succeed before App Runner decides that the service is healthy.
+        /// </summary>
+        public double? HealthCheckHealthyThreshold { get; set; }
+
+        /// <summary>
+        /// The time interval, in seconds, between health checks.
+        /// </summary>
+        public int? HealthCheckInterval { get; set; }
+
+        /// <summary>
+        /// The URL that health check requests are sent to.
+        /// </summary>
+        public string? HealthCheckPath { get; set; }
+
+        /// <summary>
+        /// The IP protocol that App Runner uses to perform health checks for your service.
+        /// </summary>
+        public string HealthCheckProtocol { get; set; } = "TCP";
+
+        /// <summary>
+        /// The time, in seconds, to wait for a health check response before deciding it failed.
+        /// </summary>
+        public int? HealthCheckTimeout { get; set; }
+
+        /// <summary>
+        /// The number of consecutive checks that must fail before App Runner decides that the service is unhealthy.
+        /// </summary>
+        public int? HealthCheckUnhealthyThreshold { get; set; }
+
+        /// <summary>
+        /// The number of CPU units reserved for each instance of your App Runner service.
+        /// </summary>
+        public string Cpu { get; set; }
+
+        /// <summary>
+        /// The amount of memory, in MB or GB, reserved for each instance of your App Runner service.
+        /// </summary>
+        public string Memory { get; set; }
```

Comment:
We are using double instead of string for Cpu and Memory

Reply:
Also, does CPU and Memory not have a defaults from the CDK side? We are marking them are nullable in other recipes. So if they have CDK defaults, lets stay consistent and mark them as nullable.

Reply:
For AppRunner Cpu and Memory are strings in CloudFormation/CDK https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apprunner-service-instanceconfiguration.html#cfn-apprunner-service-instanceconfiguration-cpu.

If the recipe is defined to always have a value which get serialized into the configuration then it will never be null.